### PR TITLE
Updated devtools + use karma to replace wct

### PR DESCRIPTION
**Pull request by hans-lizihan**

### CODE REVIEW

**Comment by hans-lizihan** [comment]
@tdumitrescu can do a round of deps update for the params work?

File: karma.config.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    browserVersion: `latest`,
+    platformName: `macOS 12`,
+    'sauce:options': {
+      extendedDebugging: true,
+    },
+  },
+  sl_firefox_old: {
+    base: `SauceLabs`,
+    browserName: `firefox`,
+    browserVersion: `74`,
```
Comment (tdumitrescu) [inline karma.config.js:41]:
```suggestion
    browserVersion: `91`,
```

File: karma.config.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    browserVersion: `latest`,
+    platformName: `macOS 12`,
+    'sauce:options': {
+      extendedDebugging: true,
+    },
+  },
+  sl_chrome_old: {
+    base: `SauceLabs`,
+    browserName: `chrome`,
+    browserVersion: `76`,
```
Comment (tdumitrescu) [inline karma.config.js:23]: how about we move this up to 90
```suggestion
    browserVersion: `90`,
```

File: package.json [hunk 3]
```diff
@@ -20,10 +20,9 @@
     "eslint:fix": "eslint --fix .",
     "publish-devtools": "node scripts/publish-devtools.js",
     "test": "npm run build-test && npm run test-server && npm run test-browser-local",
-    "test-browser-local": "wct --plugin local test/browser/index.html",
-    "test-browser-p": "wct --plugin local --persistent test/browser/index.html",
-    "test-browser-sauce": "wct --plugin sauce test/browser/index.html",
-    "test-server": "NODE_ENV=test mocha --require babel-core/register test/server",
+    "test-browser-local": "SAUCE_LAB=0 karma start karma.config.js",
+    "test-browser-sauce": "SAUCE_LAB=1 karma start karma.config.js",
```
Comment (tdumitrescu) [inline package.json:24]:
```suggestion
    "test-browser-local": "SAUCE=0 karma start karma.config.js",
    "test-browser-sauce": "SAUCE=1 karma start karma.config.js",
```

File: karma.config.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
+/* eslint-env node */
+
+const path = require(`path`);
+
+process.env.CHROME_BIN = require(`puppeteer`).executablePath();
+process.env.FIREFOX_BIN = require(`playwright`).firefox.executablePath();
+const SAUCE_LAB = parseInt(process.env.SAUCE_LAB, 10) === 1;
```
Comment (tdumitrescu) [inline karma.config.js:7]:
```suggestion
const SAUCE = parseInt(process.env.SAUCE, 10) >= 1;
```

File: karma.config.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    plugins: [
+      require(`karma-mocha`),
+      require(`karma-spec-reporter`),
+      require(`karma-chrome-launcher`),
+      require(`karma-firefox-launcher`),
+      require(`karma-sauce-launcher`),
+      require(`karma-sourcemap-loader`),
+    ],
+    browsers: SAUCE_LAB ? sauceBrowsers : [`ChromeHeadless`, `Firefox`],
+    reporters: SAUCE_LAB ? [`spec`, `saucelabs`] : [`spec`],
```
Comment (tdumitrescu) [inline karma.config.js:105]:
```suggestion
    browsers: SAUCE ? sauceBrowsers : [`ChromeHeadless`, `Firefox`],
    reporters: SAUCE ? [`spec`, `saucelabs`] : [`spec`],
```

File: test/browser/index.js [hunk 6]
```diff
@@ -3,7 +3,8 @@ import sinonChai from 'sinon-chai';
 
 chai.use(sinonChai);
 
-import 'babel-polyfill';
+import 'core-js/stable';
+import 'regenerator-runtime/runtime';
```
Comment (tdumitrescu) [inline test/browser/index.js:7]: We don't care anymore about envs that don't have native async/await. What do we need to update to get rid of this?

File: test/browser/router.js [hunk 7]
```diff
@@ -38,7 +38,7 @@ describe(`Router`, function () {
     this.routerApp = document.createElement(`router-app`);
     document.body.appendChild(this.routerApp);
 
-    await nextAnimationFrame();
+    return nextAnimationFrame();
```
Comment (tdumitrescu) [inline test/browser/router.js:41]: weird that this makes a difference? also if you do that then the function shouldn't be marked `async` anymore

Comment (hans-lizihan) [reply test/browser/router.js]: oh this should be a mistake, reverted

File: test/browser/index.js (hunk 6 above)
Comment (hans-lizihan) [reply test/browser/index.js]: removed

File: .babelrc [hunk 8]
```diff
@@ -1,3 +1,12 @@
 {
-  "presets": ["es2015"]
+  "presets": [
+    ["@babel/preset-env", {
+      "targets": {
+        "chrome": "90",
+        "firefox": "91",
+        "edge": "90",
+        "safari": "12"
```
Comment (tdumitrescu) [inline .babelrc:8]: below you have
```suggestion
        "safari": "13"
```

File: test/browser/index.js [hunk 9]
```diff
@@ -3,8 +3,6 @@ import sinonChai from 'sinon-chai';
 
 chai.use(sinonChai);
 
-import 'babel-polyfill';
```
Comment (tdumitrescu) [inline test/browser/index.js:6]: 🎉

**Review by tdumitrescu — APPROVED** [review]
woo thanks, just the one safari version change left

**Pull request merged**